[UX] Shorter resource syntax for `dstack apply`, `dstack offer`, and`dstack ps`

peterschmidt85 opened the pull request:
Closes #2571

peterschmidt85 committed:
[UX] Shorter resource syntax for `dstack apply`, `dstack offer`, and `dstack ps` #2571

r4victor commented:
Takes more space than before as the columns do not wrap:

Before:

```
 NAME               BACKEND         RESOURCES        PRICE   STATUS  SUBMITTED  
 fresh-eagle-1                                               done    3 days ago 
   replica=0 job=0  gcp             208xCPU,         $20.57  done    3 days ago 
                    (europe-west4)  1872GB, 8xH100                              
                                    (80GB), 100.0GB                             
                                    (disk), SPOT                                
   replica=0 job=1  gcp             208xCPU,         $20.57  done    3 days ago 
                    (europe-west4)  1872GB, 8xH100                              
                                    (80GB), 100.0GB                             
                                    (disk), SPOT                                
```

After:

```
 NAME              C…  MEMO…  GPU         DISK  PRICE         STATUS  SUBMITTED 
 fresh-eagle-1                                                done    3 days a… 
   replica=0 job…  2…  1872…  H100:80GB…  100…  $20.57 (spo…  done    3 days a… 
   replica=0 job…  2…  1872…  H100:80GB…  100…  $20.57 (spo…  done    3 days a…
```

peterschmidt85 committed:
Revert "[UX] Shorter resource syntax for `dstack apply`, `dstack offer`, and `dstack ps` #2571"

This reverts commit 440398edf4fa20e84517f3828ae6839acb70c158.

peterschmidt85 committed:
Shorter resource syntax for `dstack apply`, `dstack offer`, and `dstack ps` #2571" (second attempt)

peterschmidt85 committed:
Shorter resource syntax for dstack apply, dstack offer, and dstack ps #2571" (polishing)

peterschmidt85 commented:
@r4victor 
Yes, indeed, many columns don't work with narrow terminal width.
- Changed back to one RESOURCES column.
- Updated the format of RESOURCES for higher readability/ergonomics.
